src/converters/app-converter.ts

```diff
@@ -0,0 +1,388 @@
+        for (const d of dependencies) {
+            if (d instanceof FileAssetManifestConverter) {
+                this.resources.set(d.id, { resource: d.file, resourceType: d.resourceType });
+                dependsOn.push(d.file);
+            }
+            // TODO: handle docker images
+        }
+        return dependsOn;
+    }
+
+    private mapParameter(logicalId: string, typeName: string, defaultValue: any | undefined) {
+        // TODO: support arbitrary parameters?
+
+        if (!typeName.startsWith('AWS::SSM::Parameter::')) {
+            throw new Error(`unsupported parameter ${logicalId} of type ${typeName}`);
+        }
+        if (defaultValue === undefined) {
+            throw new Error(`unsupported parameter ${logicalId} with no default value`);
+        }
+
+        function parameterValue(parent: pulumi.Resource): any {
+            const key = defaultValue;
+            const paramType = typeName.slice('AWS::SSM::Parameter::'.length);
+            if (paramType.startsWith('Value<')) {
+                const type = paramType.slice('Value<'.length);
+                if (type.startsWith('List<') || type === 'CommaDelimitedList>') {
+                    return getSsmParameterList({ name: key }, { parent }).then((v) => v.value);
+                }
+                return getSsmParameterString({ name: key }, { parent }).then((v) => v.value);
+            }
+            return key;
+        }
+
+        this.parameters.set(logicalId, parameterValue(this.host));
+    }
+
+    private mapResource(
+        logicalId: string,
+        typeName: string,
+        props: any,
+        options: pulumi.ResourceOptions,
+    ): ResourceMapping {
+        if (this.host.options?.remapCloudControlResource !== undefined) {
+            const res = this.host.options.remapCloudControlResource(logicalId, typeName, props, options);
+            if (res !== undefined) {
+                debug(`remapped ${logicalId}`);
+                return res;
+            }
+        }
+
+        const awsMapping = mapToAwsResource(logicalId, typeName, props, options);
+        if (awsMapping !== undefined) {
+            debug(`mapped ${logicalId} to classic AWS resource(s)`);
+            return awsMapping;
+        }
+
+        return mapToCfnResource(logicalId, typeName, props, options);
+    }
+
+    private processOptions(resource: CloudFormationResource, parent: pulumi.Resource): pulumi.ResourceOptions {
+        const dependsOn = getDependsOn(resource);
+        return {
+            parent: parent,
+            dependsOn: dependsOn !== undefined ? dependsOn.map((id) => this.resources.get(id)!.resource) : undefined,
+        };
+    }
+
+    /** @internal */
+    asOutputValue<T>(v: T): T {
+        const value = this.host.stack.resolve(v);
+        return this.processIntrinsics(value) as T;
+    }
+
+    private processIntrinsics(obj: any): any {
+        debug(`Processing intrinsics for ${JSON.stringify(obj)}`);
```

Comment:
Should we do error handling here just to be safe? JSON.stringify can throw if there's circular refs or a BigInt